### ISSUE TITLE
quad_aspect_ratio is wrong for concave quads

### ISSUE THREAD

**Issue by bender10** [sandialabs/verdict#2]
https://github.com/sandialabs/verdict/blob/fb582cf4fc04ecbd2bf423554325bda01231227a/V_QuadMetric.cpp#L464

The `quad_aspect_ratio` function relies on the fact that the cross product of two 2d vectors happens to be the area of the parallelogram ( https://stackoverflow.com/a/243977/122207 ). One of the two areas should be negative for concave shapes when evaluated at a particular vertex. Therefore the impact is to half of all possible evaluations of quads which have a maximum angle over 180 degrees.

The VerdictVector `operator*` returns a new instance of VerdictVector and VercitVector's `length` function always returns the positive magnitude. Coupled together, the sign of the area from the cross product is lost and the final result is wrong when the quad shapes become convex.

**Comment by bender10** [comment]
The best fix here is to rely on the `signed_corner_area` function as done in other functions.

The code:
```
double quad_aspect_ratio(int /*num_nodes*/, const double coordinates[][3])
{
  VerdictVector edges[4];
  make_quad_edges(edges, coordinates);

  double a1 = edges[0].length();
  double b1 = edges[1].length();
  double c1 = edges[2].length();
  double d1 = edges[3].length();

  double ma = a1 > b1 ? a1 : b1;
  double mb = c1 > d1 ? c1 : d1;
  double hm = ma > mb ? ma : mb;

  double corner_areas[4];
  signed_corner_areas(corner_areas, coordinates);

  double aspect_ratio = hm * (a1 + b1 + c1 + d1) / (corner_areas[0] + corner_areas[1] + corner_areas[2] + corner_areas[3]);
...
```

**Comment by clintonstimpson** [comment]
Hi Tim,  Thanks for finding this!
I've done a bit of testing with your suggested changes, and have looked at updating some unit tests for this as well.

**Comment by clintonstimpson** [comment]
This was fixed in 58aeef4d.
